test/tests/functional/pbs_hook_management.py

```diff
@@ -0,0 +1,241 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2019 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+import os
+import time
+import socket
+import textwrap
+import datetime
+from tests.functional import *
+from tests.functional import MGR_CMD_SET
+from tests.functional import SERVER
+from tests.functional import ATTR_h
+from tests.functional import TEST_USER
+from tests.functional import Job
+from tests.functional import JOB
+
+
+@tags('hooks', 'smoke')
+class TestHookManagement(TestFunctional):
+
+    def test_hook_00(self):
+        """
+        By creating an import hook, it executes a management hook.
+        """
+        self.logger.info("**************** HOOK START ****************")
+        # hook_name = "test_management"
+        hook_name = "management"
+        hook_msg = 'running management hook_00'
```

Comment:
It would be better if you create a function to create the hook body and pass in the hook_msg to that function.

Reply:
Yes, this makes much more sense and less duplicated code.